retry/retry.go

```diff
@@ -153,9 +146,30 @@ func (r *retrier) attempt(continueFn ContinueFn, fn Fn) error {
 			return err
 		}
 		r.metrics.errors.Inc(1)
-		curr = int64(float64(curr) * r.backoffFactor)
 	}
 	r.metrics.errorsFinal.Inc(1)
 
 	return err
 }
+
+// BackoffNanos calculates the backoff for a retry in nanoseconds.
+func BackoffNanos(
+	retry int,
+	jitter bool,
+	backoffFactor float64,
+	initialBackoff time.Duration,
+	maxBackoff time.Duration,
+) int64 {
+	backoff := initialBackoff.Nanoseconds()
+	if jitter {
+		half := backoff / 2
+		backoff = half + rand.Int63n(half)
+	}
+	if retry >= 1 {
```

Comment:
should this be before the jitter to match the previous behavior?

Reply:
yeah will move this above, although it still does not give 100% the same math as before, but closer